core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3917,6 +3921,20 @@ private void _cleanUpReleaseAllLoggers(List<Throwable> errors) {
         }
     }
 
+    private void _cleanUpCancelDependencyGraphCalculation() {
+        LOGGER.log(Level.FINE, "Canceling internal dependency graph calculation");
+        if (scheduledFutureDependencyGraph != null && !scheduledFutureDependencyGraph.isDone()) {
```

Comment:
I think you have to syncronize the entire block because scheduledFutureDependency graph can become null in between the nullcheck and the isDone call

Reply:
I fixed it. Thank you for your feedback :)